### PR TITLE
1.20 & 1.20.1

### PR DESCRIPTION
**Pull request by PondWader**
1.20.1 is just bug fixes that don't affect the data I believe so just changing version.json was needed for that (hopefully unless Mojang did something weird, I didn't thoroughly compare 1.20.1 data against 1.20)

### CODE REVIEW

**Comment by andriycraft** [comment]
👍🏼

**Review by andriycraft — APPROVED** [review]

**Comment by rom1504** [comment]
1. What scripts and tools did you run
2. How did you validate the data is correct
3. For protocol.json how did you made the changes and from what source of information?

**Comment by frej4189** [comment]
Protocol seems okay at first glance. May take a closer look later.

Data does not seem correct. How was this generated?

**Comment by PondWader** [comment]
> 1. What scripts and tools did you run
> 2. How did you validate the data is correct
> 3. For protocol.json how did you made the changes and from what source of information?

I used Burger and the burger-data-extractor for some of it, but Burger doesn't seem to work for a lot of 1.20, so then I used the minecraft-data-generator-server and for the loot tables I used the jar-extractor repo.

I'm not too sure how to validate it, what can I validate it against?

For protocol.json I just used wiki.vg

**Comment by PondWader** [comment]
> Protocol seems okay at first glance. May take a closer look later.
> 
> Data does not seem correct. How was this generated?

Which data seems incorrect? All of it?

**Comment by frej4189** [comment]
> > Protocol seems okay at first glance. May take a closer look later.
> > 
> > Data does not seem correct. How was this generated?
> 
> Which data seems incorrect? All of it?

The presence of material.json leads me to believe the extraction method is flawed. Might just be the material related data that is incorrect. Haven't dug that deep into the files.

**Comment by PondWader** [comment]
> > > Protocol seems okay at first glance. May take a closer look later.
> > > Data does not seem correct. How was this generated?
> > 
> > 
> > Which data seems incorrect? All of it?
> 
> The presence of material.json leads me to believe the extraction method is flawed. Might just be the material related data that is incorrect. Haven't dug that deep into the files.

If you're referring to the fact Material class was removed, mc-data-generator-server doesn't seem to actually use the removed Material class to generate that data, it did however use the Material class for something to do with gourds and one other item which I had to modify although I did just make the change in GitPod and may have done something wrong.

**Comment by frej4189** [comment]
> If you're referring to the fact Material class was removed, mc-data-generator-server doesn't seem to actually use the removed Material class to generate that data, it did however use the Material class for something to do with gourds and one other item which I had to modify although I did just make the change in GitPod and may have done something wrong.

It's not that the Material class is removed. It's the entire concept of materials that is removed in favour of tags and more verbose block declarations.

**Comment by PondWader** [comment]
> It's not that the Material class is removed. It's the entire concept of materials that is removed in favour of tags and more verbose block declarations.

So do you think materials.json should just be removed? That data was extracted from the 1.20 server

**Comment by frej4189** [comment]
> > It's not that the Material class is removed. It's the entire concept of materials that is removed in favour of tags and more verbose block declarations.
> 
> So do you think materials.json should just be deleted? That data was generated from the 1.20 server

Not necessarily. Just need someone to verify the generator server does indeed infer the materials (which is now plainly a prismarine construct) correct. 

Make a PR on the generator server repo with your updates

**Comment by rom1504** [comment]
Can you post the diff between 1.19.4 and 1.20 protocol as a comment so
people may verify against wiki.vg and the decompiled code

**Comment by rom1504** [comment]
Another thing you can do for validation is open a PR in
node-minecraft-protocol similar to the 1.19.4 one that will use this mc
data PR to see if the tests pass.
Then same for mineflayer

File: data/pc/1.20/protocol.json
Comment (PondWader) [inline data/pc/1.20/protocol.json:1]: Protocol.json diff between 1.19.4 and 1.20

```diff
diff --git "minecraft-data\\data\\pc\\1.19.4\\protocol.json" "minecraft-data\\data\\pc\\1.20\\protocol.json"
index 057a5ab..780374f 100644
--- "minecraft-data\\data\\pc\\1.19.4\\protocol.json"
+++ "minecraft-data\\data\\pc\\1.20\\protocol.json"
@@ -1951,6 +1951,10 @@
                                   ]
                                 }
                               ]
+                            },
+                            {
+                              "name": "sendsTelemtryData",
+                              "type": "bool"
                             }
                           ]
                         ]
@@ -2330,10 +2334,6 @@
                 ]
               ]
             },
-            {
-              "name": "suppressLightUpdates",
-              "type": "bool"
-            },
             {
               "name": "records",
               "type": [
@@ -2700,10 +2700,6 @@
                 }
               ]
             },
-            {
-              "name": "trustEdges",
-              "type": "bool"
-            },
             {
               "name": "skyLightMask",
               "type": [
@@ -2864,10 +2860,6 @@
               "name": "chunkZ",
               "type": "varint"
             },
-            {
-              "name": "trustEdges",
-              "type": "bool"
-            },
             {
               "name": "skyLightMask",
               "type": [
@@ -3033,6 +3025,10 @@
                   ]
                 ]
               ]
+            },
+            {
+              "name": "portalCooldown",
+              "type": "varint"
             }
           ]
         ],
@@ -3352,6 +3348,10 @@
             {
               "name": "location",
               "type": "position"
+            },
+            {
+              "name": "isFrontText",
+              "type": "bool"
             }
           ]
         ],
@@ -3477,10 +3477,6 @@
             {
               "name": "duration",
               "type": "varint"
-            },
-            {
-              "name": "entityId",
-              "type": "i32"
             }
           ]
         ],
@@ -3495,10 +3491,6 @@
               "name": "playerId",
               "type": "varint"
             },
-            {
-              "name": "entityId",
-              "type": "i32"
-            },
             {
               "name": "message",
               "type": "string"
@@ -4115,6 +4107,10 @@
                   ]
                 ]
               ]
+            },
+            {
+              "name": "portalCooldown",
+              "type": "varint"
             }
           ]
         ],
@@ -6478,6 +6474,10 @@
               "name": "location",
               "type": "position"
             },
+            {
+              "name": "isFrontText",
+              "type": "bool"
+            },
             {
               "name": "text1",
               "type": "string"
```

**Comment by PondWader** [comment]
> Can you post the diff between 1.19.4 and 1.20 protocol as a comment so people may verify against wiki.vg and the decompiled code
Done: https://github.com/PrismarineJS/minecraft-data/pull/721/files#r1230896977
I'll test it on NMP soon

**Comment by frej4189** [comment]
Protocol is good

File: data/dataPaths.json [hunk 1]
```diff
@@ -1200,6 +1200,55 @@
...
+      "language": "pc/1.20",
+      "entities": "pc/1.20",
+      "protocol": "pc/1.20",
+      "windows": "pc/1.16.1",
+      "version": "pc/1.20",
+      "foods": "pc/1.20",
+      "particles": "pc/1.20",
+      "blockLoot": "pc/1.20",
+      "entityLoot": "pc/1.20",
+      "loginPacket": "pc/1.19.2",
```
Comment (rom1504) [inline data/dataPaths.json:1224]: this is likely to be needed for mineflayer

**Comment by rom1504** [comment]
thanks

**Pull request merged**